packages/ui-components/src/components/Translation/utils.js

```diff
@@ -0,0 +1,52 @@
+import { isValidElement } from 'react';
+import { TranslatedEnum, TranslatedReferenceData, TranslatedText } from '.';
+import { getReferenceDataStringId } from '@tamanu/shared/utils/translation';
+import { getEnumPrefix, toCamelCase } from '@tamanu/shared/utils/enumRegistry';
+
+/**
+ * Given a valid React element, returns true if and only if that element is a wrapper around
+ * {@link TranslatedText}. In other words, it is an element whose root might be a `<TranslatedText>`
+ * element.
+ */
+export const isTranslatedText = element => {
+  if (!isValidElement(element)) return false;
+
+  const TRANSLATED_COMPONENT_NAMES = [
+    'TranslatedText',
```

Comment:
This is a refactor. I changed t to be a list of strings from the actual components o I didn't have to import all the components